src/YieldDistributor.sol

```diff
@@ -43,22 +44,31 @@ contract YieldDistributor is IYieldDistributor, Ownable2StepUpgradeable, VotingM
     address[] public queuedProjectsForAddition;
     /// @notice Array of projects queued for removal from the next cycle
     address[] public queuedProjectsForRemoval;
-    /// @notice The voting power allocated to projects by voters in the current cycle
+    /// @notice The voting power allocated to each project by voters in the current cycle
     uint256[] public projectDistributions;
     /// @notice The last block number in which a specified account cast a vote
     mapping(address => uint256) public accountLastVoted;
-    /// @notice The voting power allocated to projects by voters in the current cycle
+    /// @notice The voting power allocated to each project by a specific voter in the current cycle
     mapping(address => uint256[]) voterDistributions;
     /// @notice How much of the yield is divided equally among projects
     uint256 public yieldFixedSplitDivisor;
     /// @notice The address of the `ButteredBread` token contract
     ERC20VotesUpgradeable public BUTTERED_BREAD;
     /// @notice The block number before the last yield distribution
     uint256 public previousCycleStartingBlock;
+    /// @notice Mapping of user addresses to their current multiplier factor
+    mapping(address => uint256) public userToMultiplier;
+    /// @notice Mapping of user addresses to their multiplier validity period
+    mapping(address => uint256) public userToValidity;
+    /// @notice The maximum multiplier incrementation
+    uint256 public maxVotingStreakMultiplier;
+    /// @notice The increment value for the multiplier
+    uint256 public votingStreakMultiplierIncrement;
 
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor() {
+    constructor(address _votingStreakMultiplier) {
```

Comment:
we should need to add them through a setter, not through the constructor 

also because these contracts are upgradeable, we actually don't use the constructor at all 

https://docs.openzeppelin.com/upgrades-plugins/writing-upgradeable